test/unit/test_reader.cpp

```diff
@@ -0,0 +1,29 @@
+#include "../include/readers.hpp"
+#include <filesystem>
+#include <gtest/gtest.h>
+#include <string>
+#include <vector>
+
+TEST(ReadersTest, CSV) {
+
+  std::filesystem::path source_file = __FILE__;
+  std::filesystem::path source_dir = source_file.parent_path();
+  std::filesystem::path test_resources_dir = source_dir / "../test_resources";
+  std::filesystem::path csv_test_file =
+      test_resources_dir / "charge_exchange_h0_h1.csv";
+
+  std::vector<std::vector<double>> all_data =
+      CSVReader(std::string(csv_test_file)).GetData();
+  std::vector<double> rates = CSVReader(std::string(csv_test_file)).GetRates();
+  std::vector<double> temps = CSVReader(std::string(csv_test_file)).GetTemps();
+
+  ASSERT_NEAR(all_data[0][0], 1.00, 1.01);
```

Comment:
Last arg to ASSERT_NEAR should be the tolerance when comparing the first two args for equality. I'd expect the values to be equal to the precision as the string val, so hopefully this should work:
```suggestion
  ASSERT_NEAR(all_data[0][0], 1.0067508210214045, 1e-16);
```